plugins/woocommerce/src/Caching/ObjectCache.php

```diff
@@ -286,41 +258,16 @@ public function get( $id, int $expiration = self::DEFAULT_EXPIRATION, callable $
 	 * @return bool True if the object is removed from the cache successfully, false otherwise (because the object wasn't cached or for other reason).
 	 */
 	public function remove( $id ): bool {
-		$result = $this->get_cache_engine()->delete_cached_object( $this->get_cache_key_prefix() . $id );
-
-		/**
-		 * Action triggered by an instance of ObjectCache after an object is (attempted to be) removed from the cache.
-		 *
-		 * @since 6.8.0
-		 *
-		 * @param int|string $id The id of the object being removed.
-		 * @param bool $result True if the object removal succeeded, false otherwise.
-		 */
-		do_action( "woocommerce_after_removing_{$this->object_type}_from_cache", $id, $result );
-
-		return $result;
+		return $this->get_cache_engine()->delete_cached_object( $id, $this->get_object_type() );
 	}
 
 	/**
 	 * Remove all the objects from the cache.
-	 * This is done by forcing the generation of a new cache key prefix
-	 * and leaving the old cached objects to expire.
 	 *
-	 * @return void
+	 * @return bool True on success, false on error.
 	 */
-	public function flush(): void {
-		delete_option( $this->cache_key_prefix_option_name );
-		$this->cache_key_prefix = null;
-
-		/**
-		 * Action triggered by an instance of ObjectCache after it flushes all the cached objects.
-		 *
-		 * @since 6.8.0
-		 *
-		 * @param ObjectCache $cache_instance The instance of ObjectCache whose 'flush` method has been called.
-		 * @param CacheEngine $engine The cache engine in use.
-		 */
-		do_action( "woocommerce_after_flushing_{$this->object_type}_cache", $this, $this->get_cache_engine() );
```

Comment:
Has this action been removed on purpose?

Reply:
Yeah, so now that we are using the same namespacing, this action is no longer reliable. The cache could be flushed by legacy code without it actually hitting here (and also by memcache server restart and other non capturable events). It's much more reliable to hook to order modify methods to capture cache invalidations.